presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushDownDereferences.java

```diff
@@ -401,7 +385,7 @@ protected Result pushDownDereferences(Context context, JoinNode joinNode, BiMap<
                             .addAll(leftNode.getOutputVariables())
                             .addAll(rightNode.getOutputVariables())
                             .build(),
-                    joinNode.getFilter(),
+                    joinNode.getFilter().map(expression -> replaceDereferences(expression, expressions)),
```

Comment:
why don't we have this before?

Reply:
not needed, we already have ExtractFromJoin. let me fix